daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TimeSpec.scala

```diff
@@ -255,21 +256,27 @@ class TimeSpec extends AnyFreeSpec with Inside with Matchers with TableDrivenPro
 
     "fromString functions reject non UTC ISO 8601 compliant time strings" in {
       val testCases = Table(
-        "withNanos",
+        "string",
+        "",
         "001-01-01Z",
         "0001-01-01Z",
         "1970-01-01T01",
         "2023-06-28T07:19",
         "1969-07-20T20:17:00.0000000000Z",
         "01969-07-20T20:17:00.0000000000Z",
+        "1969-13-20T20:17:00",
+        "1969-07-20T20:17:00Z ",
+        " 1969-07-20T20:17:00Z",
         "1969-13-20T20:17:00Z",
         "1969-07-32T20:17:00Z",
-        "1969-07-32T20:17:00Z−07:01",
-        "1969-07-32T20:17:00Z−0700",
-        "1969-07-32T20:17:00Z−07",
-        "1969-07-32T20:17:00Z+07:00",
-        "1969-07-32T20:17:00Z+0701",
-        "1969-07-32T20:17:00Z+07",
+        "1969-07-20T20:17:00+00:00",
+        "1969-07-20T20:17:00-00:00",
+        "1969-07-20T20:17:00−07:01",
+        "1969-07-20T20:17:00−0700",
+        "1969-07-20T20:17:00−07",
+        "1969-07-20T20:17:00+07:00",
+        "1969-07-20T20:17:00+0701",
+        "1969-07-20T20:17:00+07",
```

Comment:
I manually check that all those string were rejected by main running with java 8.